pep_sphinx_extensions/pep_processor/transforms/pep_headers.py

```diff
@@ -97,20 +98,28 @@ def apply(self) -> None:
             field.parent.remove(field)
 
 
-def _list_name_from_thread(node: nodes.reference) -> nodes.raw:
+def _pretty_thread(text: nodes.Text) -> nodes.Text:
+    # we want to keep these as lowercase:
+    if "python-dev" in text:
+        return nodes.Text("python-dev")
+    if "python-committers" in text:
+        return nodes.Text("python-committers")
+
+    parts = text.split("/")
+
     # mailman structure is
     # https://mail.python.org/archives/list/<list name>/thread/<id>
+    try:
+        list_name = parts[parts.index("archives") + 2].removesuffix("@python.org")
+        return nodes.Text(list_name.title().replace("Sig", "SIG").replace("-", " "))
```

Comment:
```suggestion
        return nodes.Text(list_name.title().replace("Sig", "SIG"))
```

IMO, I don't think we should remove dashes if present, since they are included in almost all the [canonical list names](https://mail.python.org/mailman3/lists/) and a large majority of the original list titles being replaced here.

Reply:
I approach it from the other direction -- "typing-sig" is the email list for the typing special interest group, which I would write in English as "Typing SIG" (with the definite article). So from that perspective replacing dashes makes sense.

A

Reply:
Yeah, I'm not strongly wed to this, but my point is that regardless of how you would personally write it, the hyphen is in the name of the _mailing list_ is the current status quo as used in PEPs (aside from Distutils SIG, which no longer exists), and requires additional processing and special cases that complicates the implementation (e.g. the above). Also, keep in mind these are the (canonical) names of the _mailing lists_ not the groups themselves, which the hyphen makes less ambiguous. Therefore, I again think we should just leave it as it is.

Reply:
Let's just leave the hyphens in place. It's simpler and aligned with current practice.